chef_master/source/api_chef_server.rst

```diff
@@ -712,8 +721,9 @@ with a request body similar to:
 
 where:
 
-* ``name`` must begin with a lower-case letter or digit, may only contain lower-case letters, digits, hyphens, and underscores. For example: ``chef``.
-* ``email``, ``name``, and ``password`` are all required to be present and have a value.
+* ``username`` must begin with a lower-case letter or digit, may only contain lower-case letters, digits, hyphens, and underscores. For example: ``chef``.
+* ``display_name``, ``email`` and  ``username`` are required to be present and have a value.
```

Comment:
Some additional validation is done on the email to attempt to restrict it a bit. The somewhat gruesome regex is here: https://github.com/chef/chef-server/blob/master/src/oc_erchef/apps/chef_objects/src/chef_user.erl#L224. 
Usernames are a dotted list of ```a-z0-9!#$%&'*+/=?^_`{|}~-``` elements, while the host portion can be a dotted list of '[a-z0-9](?:[a-z0-9-]*[a-z0-9])' elements, or an IPv4 address, or some other weird stuff.

Reply:
Possible text:
* ``email`` is required to be present and fit a loose definition of a valid email address. The basic form is a username@host structure, where username is a dot separated list of ```a-z0-9!#$%&'*+/=?^_`{|}~-``` elements, while the host name can be a dot separated list of ```[a-z0-9](?:[a-z0-9-]*[a-z0-9])``` elements, or an IPv4 address, or a few other things.
I'm skeptical this actually is correct in a unicode world.

Reply:
I thought about including a link to the code to document the email regex, but that seems like a maintenance nightmare.  Instead I included a simple description of the requirements.

Reply:
That seems entirely reasonable to me.